### PR TITLE
New exercise: Huffman coding.

### PR DESCRIPTION
**Pull request by fpottier**

### CODE REVIEW

File: exercises/huffman/descr.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+user-specified total order on elements) and `q'` is the queue `q` deprived of
+`x`.
+
+**Question 2.**
+Define a module `Q` that provides priority queues whose
+elements are pairs of a tree and an integer frequency. Drawing
+an element of the queue must yield an element whose frequency
+is minimal.
+Then,
+Define the function `build_tree`,
```
Comment (yurug) [inline exercises/huffman/descr.md:226]:
```suggestion
define the function `build_tree`,
```

File: exercises/huffman/descr.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+a leaf, which carries a character.
+
+**Question 4.** Write a function `find` of type
+`data -> int -> tree -> char * int`
+such that `find data i t`
+descends in the tree `t` along the path
+that is encoded at index `i` in the binary data `data`,
+until a leaf is reached,
+and returns the character carried by that leaf
+as well as an updated index into the binary data `data`.
```
Comment (yurug) [inline exercises/huffman/descr.md:267]:
```suggestion
as well as an updated index into the binary data `data`. The index corresponds to the bit in `data` that follows the encoding of `c`.
```

File: exercises/huffman/descr.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+path in the tree. By descending along this path, one reaches
+a leaf, which carries a character.
+
+**Question 4.** Write a function `find` of type
+`data -> int -> tree -> char * int`
+such that `find data i t`
+descends in the tree `t` along the path
+that is encoded at index `i` in the binary data `data`,
+until a leaf is reached,
+and returns the character carried by that leaf
```
Comment (yurug) [inline exercises/huffman/descr.md:266]:
```suggestion
and returns the character `c` carried by that leaf
```

File: exercises/huffman/descr.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+At the other end, whoever wishes to decode the data must
+first **deserialize** the decoding dictionary, that is,
+read its binary representation and convert it back to a
+tree.
+
+**Question 5.**
+Complete the definitions of the functions `write`,
+whose type is `tree -> data`,
+and `read`,
+whose type is `data -> tree * int`.
```
Comment (yurug) [inline exercises/huffman/descr.md:289]:
```suggestion
whose type is `data -> tree * int`. The returned integer is the index next to the end of the tree encoding in the input data.
```

Comment (fpottier) [reply exercises/huffman/descr.md:289]: I would rather say "The integer result returned by `read` is the end index in the input data of the serialized dictionary (i.e., it is the index of whatever data follows the serialized dictionary)."

**Comment by yurug** [comment]
Thank you!

**Pull request merged**